DQM/SiPixelPhase1Summary/src/SiPixelPhase1Summary.cc

```diff
@@ -0,0 +1,175 @@
+// -*- C++ -*-
+//
+// Package:    SiPixelPhase1Summary
+// Class:      SiPixelPhase1Summary
+// 
+/**\class 
+
+ Description: Create the Phsae 1 pixel summary map
+
+ Implementation:
+     <Notes on implementation>
+*/
+//
+// Original Author:  Duncan Leggat
+//         Created:  5th December 2016
+//
+//
+#include "DQM/SiPixelPhase1Summary/interface/SiPixelPhase1Summary.h"
+// Framework
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+// DQM Framework
+#include "DQM/SiPixelCommon/interface/SiPixelFolderOrganizer.h"
+#include "DQMServices/Core/interface/DQMStore.h"
+// Geometry
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "Geometry/TrackerGeometryBuilder/interface/PixelGeomDetUnit.h"
+#include "Geometry/CommonTopologies/interface/PixelTopology.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+// DataFormats
+#include "DataFormats/DetId/interface/DetId.h"
+#include "DataFormats/SiPixelDetId/interface/PixelSubdetector.h"
+#include "DataFormats/SiPixelDetId/interface/PixelBarrelName.h"
+#include "DataFormats/SiPixelDetId/interface/PixelBarrelNameUpgrade.h"
+#include "DataFormats/SiPixelDetId/interface/PixelEndcapName.h"
+#include "DataFormats/SiPixelDetId/interface/PixelEndcapNameUpgrade.h"
+//
+#include <string>
+#include <stdlib.h>
+#include <iostream>
+#include <fstream>
+
+using namespace std;
+using namespace edm;
+
+SiPixelPhase1Summary::SiPixelPhase1Summary(const edm::ParameterSet& iConfig) :
+  conf_(iConfig),
+  firstLumi(true)
+{
+
+   LogInfo ("PixelDQM") << "SiPixelPhase1Summary::SiPixelPhase1Summary: Got DQM BackEnd interface"<<endl;
+   topFolderName_ = conf_.getParameter<std::string>("TopFolderName"); 
+   runOnEndLumi_ = conf_.getParameter<bool>("RunOnEndLumi"); 
+   runOnEndJob_ = conf_.getParameter<bool>("RunOnEndJob");
+
+   //   summaryMapNames_ = {"Summary","Digi","ADC","NClustsTotal","NClustsOnTrk","ClustWidthOnTrk","ClustWidthOffTrk","NHits","Effic","ResidualMean","ResidualRMS"}; 
+   summaryMapNames_ = {"Summary","ADC","Digi","NClustsTotal","ClustWidthOnTrk","Charge"};
+   xAxisLabels_ = {"BMO","BMI","BPO ","BPI","HCMO_1","HCMO_2","HCMI_1","HCMI_2","HCPO_1","HCPO_2","HCPI_1","HCPI_2"};
```

Comment:
Hi @leggat -since these are hardwired anyway, just declare them where they are used. No need to be kept as member data.